providers/sftp/src/airflow/providers/sftp/sensors/sftp.py

```diff
@@ -76,14 +77,15 @@ def __init__(
         self.hook: SFTPHook | None = None
         self.sftp_conn_id = sftp_conn_id
         self.newer_than: datetime | str | None = newer_than
+        self.managed_conn = managed_conn
         self.python_callable: Callable | None = python_callable
         self.op_args = op_args or []
         self.op_kwargs = op_kwargs or {}
         self.deferrable = deferrable
 
-    def poke(self, context: Context) -> PokeReturnValue | bool:
-        self.hook = SFTPHook(self.sftp_conn_id)
-        self.log.info("Poking for %s, with pattern %s", self.path, self.file_pattern)
+    def _get_files(self):
```

Comment:
```suggestion
    def _get_files(self) -> list[Path | str]:
```

kinda not sure the type

Reply:
the type is str, done